src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
 public class WorkWithFile {
+    public static final int ZERO_NUMBER = 0;
+    public static final int MAP_INDEX_KEY = 0;
+    public static final int MAP_INDEX_VALUE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> resultFileMap = readFile(fromFileName);
+        writeFile(toFileName, resultFileMap);
+    }
+
+    public Map<String, Integer> readFile(String fileName) {
+        File fromFile = new File(fileName);
+        Map<String, Integer> fileMap = new LinkedHashMap<>();
+        fileMap.put("supply", ZERO_NUMBER);
+        fileMap.put("buy", ZERO_NUMBER);
+        fileMap.put("result", ZERO_NUMBER);
+
+        try (BufferedReader readerFrom = new BufferedReader(new FileReader(fromFile))) {
+            String value = readerFrom.readLine();
+
+            while (value != null) {
+                String[] valueStringArray = value.split(",");
+
+                if (fileMap.containsKey(valueStringArray[MAP_INDEX_KEY])) {
```

Comment:
put valueStringArray[MAP_INDEX_KEY] and valueStringArray[MAP_INDEX_VALUE] to local variables and reuse them

your code will look cleaner

Reply:
Not resolved